Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RenderPass.cpp

```diff
@@ -193,9 +193,17 @@ namespace AZ
         void RenderPass::FrameBeginInternal(FramePrepareParams params)
         {
             m_timestampResult = AZ::RPI::TimestampResult();
+
+            // the pass is potentially re-allocated amount devices dynamically in runtime so deviceIndex is updated everyframe.
+            int deviceIndex = m_deviceIndex;
+            if (deviceIndex == AZ::RHI::MultiDevice::InvalidDeviceIndex && m_parent)
+            {
+                deviceIndex = azrtti_cast<AZ::RPI::ParentPass*>(m_parent)->GetDeviceIndex();
+            }
+
             if (GetScopeId().IsEmpty())
             {
-                InitScope(RHI::ScopeId(GetPathName()), m_hardwareQueueClass, m_deviceIndex);
+                InitScope(RHI::ScopeId(GetPathName()), m_hardwareQueueClass, deviceIndex);
```

Comment:
Remove the deviceIndex variable and simply call GetDeviceIndex() (RenderPass inherits from Pass)?

Reply:
Same reason as the comment before. I am adding another commit to this PR. Now the `ScopeProducer::m_deviceIndex` is set from pass data and initialized by user, `Pass::m_deviceIndex` is inheriting from its parent pass.

Reply:
This is not how it's supposed to work.